scenes/menu/main_menu.gd

```diff
@@ -315,3 +374,25 @@ func _on_bgm_check_box_toggled(button_pressed : bool):
 		start_music()
 	else:
 		stop_music()
+
+func _on_players_button_pressed():
+	modal_list.show_player_list()
+
+func _on_matches_button_pressed():
+	modal_list.show_match_list()
+
+
+func _on_modal_list_join_match_pressed(row_index):
+	var matches = NetworkManager.get_match_list()
+	var selected_match = matches[row_index]
+	room_select.text = selected_match['name']
+	_on_join_button_pressed()
+
+func _on_modal_list_observe_match_pressed(row_index):
+	var matches = NetworkManager.get_match_list()
+	var selected_match = matches[row_index]
+	var room_name = selected_match['name']
+	var player_name = get_player_name()
+	NetworkManager.observe_room(player_name, room_name)
+	update_buttons(true)
```

Comment:
Would it be too much extra effort to support people opting out of being observed? (e.g. players can press a checkbox before entering a room/matchmaking; if either player has it enabled then the match object stores a flag that this checks)

Reply:
do we really need that? I don't think I would allow it for the matchmaking queue. And nobody really even uses rooms.

Reply:
I do see a fair number of rooms occasionally, but yee fair enough